runner/runner.go

```diff
@@ -252,7 +339,7 @@ func (r *Runner) Wait() error {
 
 	var wg sync.WaitGroup
 
-	for poolId, repo := range r.repositories {
+	for poolId, repo := range r.poolManagerCtrl.GetRepoPoolManagers() {
```

Comment:
```GetRepoPoolManagers()``` should also return an error, unless it's absolutely guaranteed to never fail under any circumstances.

Reply:
I commented above why this is guaranteed to never fail. Let me know if that sounds good to you.